src/components/components/ebay-tooltip-base/index.js

```diff
@@ -0,0 +1,173 @@
+const Expander = require('makeup-expander');
+const emitAndFire = require('../../../common/emit-and-fire');
+const template = require('./template.marko');
+
+function getInitialState(input) {
+    input.location = input.location || 'bottom';
+
+    return input;
+}
+
+function getTemplateData(state) {
+    return state;
+}
+
+function init() {
+    const isHostPresent = this.el.querySelector(`.${this.state.type}__host`);
+
+    if (isHostPresent) {
+        this.expander = new Expander(this.el, {
+            hostSelector: `.${this.state.type}__host`,
+            contentSelector: `.${this.state.type}__overlay`,
+            focusManagement: this.state.type === 'tourtip' ? null : 'focusable',
+            expandOnFocus: this.state.type === 'tooltip',
+            expandOnHover: this.state.type === 'tooltip',
+            expandOnClick: this.state.type === 'infotip',
+            autoCollapse: this.state.type === 'tooltip'
+        });
+    }
+
+    this.alignOverlay();
+}
+
+function handleExpand() {
+    this.alignOverlay();
+    this.setState('expanded', true);
+    emitAndFire(this, 'tooltip-expand');
+}
+
+function handleCollapse() {
+    this.setState('expanded', false);
+    emitAndFire(this, 'tooltip-collapse');
+}
+
+function handleTooltipClose() {
+    this.setState('expanded', false);
+    this.expander.collapse();
+    emitAndFire(this, 'tooltip-close');
+}
+
+function alignOverlay() {
+    const overlay = this.el.querySelector(`.${this.state.type}__overlay`);
+
+    if (!overlay) {
+        return;
+    }
+
+    if (this.state.styleTop || this.state.styleLeft || this.state.styleRight || this.state.styleBottom) {
```

Comment:
Is overriding the positioning something we currently have a use case for? I feel that we have covered the cases we need too and that exposing this API allows users to deviate from the DS and is a little clunky.

Reply:
This is really for fine tuning. We could remove it, but I'm 100% sure developers will ask for it.